tests/next_tests/integration_tests/feature_tests/ffront_tests/test_program.py

```diff
@@ -36,205 +40,260 @@
+
+    cases.verify(
+        cartesian_case,
+        prog,
+        a,
+        b,
+        out_a,
+        out_b,
+        inout=(out_a.array()[1:], out_b.array()[1:]),
+        ref=(a.array()[1:], b.array()[1:]),
+        comparison=lambda a, b: (
+            np.all(a[1:] == out_a[1:]),
+            np.all(b[1:] == out_b[1:]),
+            out_a[0] == 0.0,
+            out_b[0] == 0.0,
+        ),
+    )
 
-def test_tuple_program_return_constructed_inside_nested(fieldview_backend):
-    size = 10
-    a = np_as_located_field(IDim)(np.ones((size,)))
-    b = np_as_located_field(IDim)(2 * np.ones((size,)))
-    c = np_as_located_field(IDim)(3 * np.ones((size,)))
-    out_a = np_as_located_field(IDim)(np.zeros((size,)))
-    out_b = np_as_located_field(IDim)(np.zeros((size,)))
-    out_c = np_as_located_field(IDim)(np.zeros((size,)))
 
-    @field_operator
+def test_tuple_program_return_constructed_inside_nested(cartesian_case):
+    @gtx.field_operator
     def pack_tuple(
-        a: Field[[IDim], float64], b: Field[[IDim], float64], c: Field[[IDim], float64]
-    ) -> tuple[tuple[Field[[IDim], float64], Field[[IDim], float64]], Field[[IDim], float64]]:
+        a: cases.IFloatField, b: cases.IFloatField, c: cases.IFloatField
+    ) -> tuple[tuple[cases.IFloatField, cases.IFloatField], cases.IFloatField]:
         return ((a, b), c)
 
-    @program(backend=fieldview_backend)
+    @gtx.program
     def prog(
-        a: Field[[IDim], float64],
-        b: Field[[IDim], float64],
-        c: Field[[IDim], float64],
-        out_a: Field[[IDim], float64],
-        out_b: Field[[IDim], float64],
-        out_c: Field[[IDim], float64],
+        a: cases.IFloatField,
+        b: cases.IFloatField,
+        c: cases.IFloatField,
+        out_a: cases.IFloatField,
+        out_b: cases.IFloatField,
+        out_c: cases.IFloatField,
     ):
         pack_tuple(a, b, c, out=((out_a, out_b), out_c))
 
-    prog(a, b, c, out_a, out_b, out_c, offset_provider={})
-
-    assert np.allclose(a, out_a)
-    assert np.allclose(b, out_b)
+    a = cases.allocate(cartesian_case, prog, "a").strategy(cases.ConstInitializer(1))()
+    b = cases.allocate(cartesian_case, prog, "b").strategy(cases.ConstInitializer(2))()
+    c = cases.allocate(cartesian_case, prog, "b").strategy(cases.ConstInitializer(3))()
+    out_a = cases.allocate(cartesian_case, prog, "out_a").strategy(cases.ConstInitializer(0))()
+    out_b = cases.allocate(cartesian_case, prog, "out_b").strategy(cases.ConstInitializer(0))()
+    out_c = cases.allocate(cartesian_case, prog, "out_c").strategy(cases.ConstInitializer(0))()
+
+    cases.verify(
+        cartesian_case,
+        prog,
+        a,
+        b,
+        c,
+        out_a,
+        out_b,
+        out_c,
+        inout=((out_a.array(), out_b.array()), out_c.array()),
+        ref=((a.array(), b.array()), c.array()),
+        comparison=lambda a, b: (np.all(a == out_a), np.all(b == out_b)),
```

Comment:
The comparison must return a single boolean value. A non-empty container of bool values is itself always `True`-like in Python (this is also the reason why `numpy.all()` exists in the first place). Examples:

```python
In [3]: bool((False, False))
Out[3]: True
```

```python
[ins] In [2]: assert (False, False)
<>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
<ipython-input-2-f321c071ff0c>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (False, False)
```

Reply:
Maybe to avoid accidentally always-true tests, `verify` should assert that the result of the comparison is a `bool` instance?